### PR TITLE
Module & Theme less optimization for cross browser compatibility.

### LINKED ISSUE #30729: [Issue] Module & Theme less optimization for cross browser compatibility.
This issue is automatically created based on existing pull request: magento/magento2#30728: Module & Theme less optimization for cross browser compatibility.

---------

### Description (*)

Updated core modules & theme less design with Magento default Flex layout less mixins. Reference taken from `<Magento2>/lib/web/css/source/lib/_utilities.less:409 Flex layout mixins` and my Magento 2 theme development experience. Using these provided mixins, the design will work on cross-browser platform properly. I've also removed some redundant code from some direct less styles, as they were already been covered by less mixins.

### Related Pull Requests

N/A

### Fixed Issues (if relevant)

N/A

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by pratiksangani**
### Description (*)

Updated core modules & theme less design with Magento default Flex layout less mixins. Reference taken from `<Magento2>/lib/web/css/source/lib/_utilities.less:409 Flex layout mixins` and my Magento 2 theme development experience. Using these provided mixins, the design will work on cross-browser platform properly. I've also removed some redundant code from some direct less styles, as they were already been covered by less mixins.

### Related Pull Requests

N/A

### Fixed Issues (if relevant)

N/A

### Manual testing scenarios (*)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#30729: Module & Theme less optimization for cross browser compatibility.

### CODE REVIEW

**Comment by pratiksangani** [comment]
@ihor-sviziev Will you please proceed this PR?

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @pratiksangani,
Could you fix failing static tests?

File: app/design/adminhtml/Magento/backend/Magento_Customer/web/css/source/_module.less [hunk 1]
```diff
@@ -28,8 +28,8 @@
     }
 
     .customer-default-address-wrapper {
-        align-items: flex-start;
-        display: flex;
+        .lib-vendor-box-align(flex-start);
+        .lib-vendor-prefix-display(flex);
         float: left;
```
Comment (mrtuvn) [inline app/design/adminhtml/Magento/backend/Magento_Customer/web/css/source/_module.less:33]: default param .lib-vendor-prefix-display() is flex so you don't need to more specific except you want to try inline flex such as .lib-vendor-prefix-display(inline-flex)

File: app/design/adminhtml/Magento/backend/web/css/source/_actions.less [hunk 2]
```diff
@@ -438,8 +438,6 @@ button {
         &._parent {
             .lib-vendor-prefix-display(flex);
             .lib-vendor-prefix-flex-direction(row);
-            display: flex;
-            flex-direction: row;
```
Comment (mrtuvn) [inline app/design/adminhtml/Magento/backend/web/css/source/_actions.less:441]: we don't have to specific flex in here when call .lib-vendor-prefix

**Review by mrtuvn — CHANGES_REQUESTED** [review]

File: app/design/adminhtml/Magento/backend/web/css/source/_actions.less (hunk 2 above)
Comment (mrtuvn) [reply app/design/adminhtml/Magento/backend/web/css/source/_actions.less:441]: And the rest changes fine with me

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8416](https://jira.corp.magento.com/browse/ENGCOM-8416) has been created to process this Pull Request

**Comment by digiltd** [comment]
Is there still a need to prefix all the flex stuff these days?

Granted there will be some cases where IE11 or Safari installed before 2015 needs to be supported, though if you are building for those browsers, flexbox prefixes are the least of your concerns :)

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Backwards_Compatibility_of_Flexbox#Status_in_browsers

**Comment by ihor-sviziev** [comment]
@digiltd good point! 
@pratiksangani could you explain which compatibility you would like to fix? Seems like we safely could remove these vendor prefixes

**Comment by digiltd** [comment]
@ihor-sviziev I think `appearance` still warrants a prefix, though only `-webkit-` and `-moz-` as Edge and IE use the `-webkit-` prefix

https://caniuse.com/css-appearance

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @pratiksangani,
Could you revert all changes and leave the only changes for `appearance`?

**Comment by ihor-sviziev** [comment]
@pratiksangani  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**